gateway/src/test/java/org/georchestra/gateway/filter/headers/providers/GeorchestraUserHeadersContributorTest.java

```diff
@@ -85,7 +88,10 @@ void testNoUser() {
 
         HttpHeaders target = new HttpHeaders();
         contributor.accept(target);
-        assertTrue(target.isEmpty());
+        assertFalse(target.isEmpty());
```

Comment:
Changing a `assertTrue` to `assertFalse` to fix the testsuite is rather scary, to be honest. 

Can you explain why such a modification is needed now ? Is it due to the modifications introduced by this PR ?

Reply:
Reading the rest of the test gives a bit more context, but then I don't get why we need to send back the `sec-external-user` header, whatever the value, if the user is not connected. I think we are better off not sending the header at all in this case, and then the assertTrue sounds perfectly legit to me.

Reply:
PR updated and new PR was created to handle NPE : 

https://github.com/georchestra/georchestra/pull/4203